openwisp_controller/config/api/serializers.py

```diff
@@ -149,6 +151,16 @@ def _create_config(self, device, config_data):
             raise serializers.ValidationError({'config': error.messages})
 
     def _update_config(self, device, config_data):
+        if (
+            config_data.get('backend') == app_settings.BACKENDS[0][0]
+            and not config_data.get('templates')
+            and not config_data.get('context')
+            and not config_data.get('config')
```

Comment:
Can you get the default values from the serializer fields and compare each property to the corresponding serializer field default? Such a solution would be more robust to future changes, I doubt anyone changing the default values (if ever needed, but we never know) of the serializer or model will remember to change this.

Reply:
no problem let's use the model then